tests/qchem/of_tests/test_molecular_dipole.py

```diff
@@ -0,0 +1,419 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for molecular dipole.
+"""
+# pylint: disable=too-many-arguments, protected-access
+import pytest
+
+import pennylane as qml
+from pennylane import I, X, Y, Z
+from pennylane import numpy as np
+from pennylane import qchem
+
+h2 = ["H", "H"]
+x_h2 = np.array([0.0, 0.0, -0.661, 0.0, 0.0, 0.661])
+coeffs_h2 = []
+coeffs_h2.append([0.0])
+coeffs_h2.append([0.0])
+coeffs_h2.append([0.45445016, 0.45445016, 0.45445016, 0.45445016])
+
+
+ops_h2 = []
+ops_h2.append([I(0)])
+ops_h2.append([I(0)])
+ops_h2.append(
+    [
+        Y(0) @ Z(1) @ Y(2),
+        X(0) @ Z(1) @ X(2),
+        Y(1) @ Z(2) @ Y(3),
+        X(1) @ Z(2) @ X(3),
+    ]
+)
+
+coeffs_h2_parity = []
+coeffs_h2_parity.append([0.0])
+coeffs_h2_parity.append([0.0])
+coeffs_h2_parity.append([-0.45445016, -0.45445016, -0.45445016, -0.45445016])
+
+
+ops_h2_parity = []
+ops_h2_parity.append([I(0)])
+ops_h2_parity.append([I(0)])
+ops_h2_parity.append(
+    [
+        Y(0) @ Y(1),
+        X(0) @ X(1) @ Z(2),
+        Y(1) @ Y(2),
+        Z(0) @ X(1) @ X(2) @ Z(3),
+    ]
+)
+
+eig_h2 = []
+eig_h2.append([0.0, 0.0])
+eig_h2.append([0.0, 0.0])
+eig_h2.append([-1.81780062, -0.90890031, -0.90890031])
+
+
+h3p = ["H", "H", "H"]
```

Comment:
Are these the same tests as in other original functions?

Reply:
These tests are mostly same as from the original functions, I have added a couple of more tests, one for wires mapping, also for parity mapping in both cases.